build/doxygen/doxygen.bzl

```diff
@@ -0,0 +1,109 @@
+"""Repository rule for building doxygen from source with libclang support.
+
+This rule downloads and builds doxygen from source using CMake, linking against
+the system LLVM/clang installation to enable clang-assisted parsing of C++ code.
+"""
+
+def _doxygen_repository_impl(ctx):
```

Comment:
Wot in tarnation... is there really no easier way to build Doxygen from Bazel?

Reply:
I have no idea. I didn't try lol. I *really* dislike dealing with build tooling so went with the easiest thing that could possibly work

Reply:
I had tried that approach initially but kept running into issues with the clang support... building the doxygen here was the only way I was able to get it to work reasonably with a couple of the other updates.